### PR TITLE
 add more types parsing for string coerceValueIntoField

### PR DESCRIPTION
**Pull request by katewang0710**
Hi,

change code to make coerceValueIntoField support more type parsing for string type.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Can you handle the primitive types as well?
Also, I think the correct way to check is `if (fieldType.isAssgnableFrom(<type>.class))`

**Comment by katewang0710** [comment]
Hi @Jackie-Jiang,

Do you mean support  boolean , byte , char , short , int , long , float and double for string type value?

Thanks,
Kate

**Comment by Jackie-Jiang** [comment]
@katewang0710 Yes, similar to how we handle primitive int and boolean

**Comment by katewang0710** [comment]
@kishoreg @Jackie-Jiang Looks like all the tests have passed. Can we merge this change? If anything needs to change please let me know.

Thanks

File: pinot-common/src/main/java/org/apache/pinot/common/config/Deserializer.java [hunk 1]
```diff
@@ -349,17 +349,26 @@
...
-            } else {
-              // TODO Add float/double/short/char
+            } else if (fieldType.isAssignableFrom(Float.class) || fieldType.isAssignableFrom(float.class)){
+              destinationValue = Float.parseFloat(stringValue);
+            } else if (fieldType.isAssignableFrom(Double.class) || fieldType.isAssignableFrom(double.class)){
+              destinationValue = Double.parseDouble(stringValue);
+            } else if (fieldType.isAssignableFrom(Short.class) || fieldType.isAssignableFrom(short.class)){
+              destinationValue = Short.parseShort(stringValue);
+            } else if (fieldType.isAssignableFrom(Byte.class) || fieldType.isAssignableFrom(byte.class)){
+              destinationValue = stringValue.getBytes()[0];
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/config/Deserializer.java]: Byte.parseByte(stringValue)?

Comment (katewang0710) [reply pinot-common/src/main/java/org/apache/pinot/common/config/Deserializer.java]: Exactly! Thanks

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/config/Deserializer.java]: Can you fix the format issue? Then I'll merge it.

**Pull request merged**